kernel/os/src/os_mempool.c

```diff
@@ -90,7 +90,7 @@ os_mempool_init(struct os_mempool *mp, int blocks, int block_size,
     struct os_memblock *block_ptr;
 
     /* Check for valid parameters */
-    if (!mp || (blocks < 0) || (block_size <= 0)) {
+    if (!mp || (blocks < 0) || (block_size < sizeof(struct os_memblock))) {
```

Comment:
Hi Marco, I saw a comment on #700 but frankly speaking I don't understand why we need this change.
As I understand `void *membuf`already has already is big enough if OS_MEMPOOL_BYTES() was used to calculate needed memory. Maybe could you say where is the issue you are seeing? I must miss something.

Reply:
Hi! Issue would appear if the user did not use OS_MEMPOOL_BYTES(), and specified block_size of, say 2 bytes. Clearly the test against > 0 is not correct, and we should have at least a pointer's worth of memory on every block.
As you have seen from the code, we typecast the beginning of a block into struct os_memblock and link it through that struct.

I'd be ok with removing the argument checks altogether, in which case the responsibility of passing correct arguments is up to the user. But if we keep the checks, then they should be meaningful.

Hope this helps,
M

Reply:
IMO block_size<=0 check is fine since this checks size of block as seen for application, not as it is required internally. We may add note to documentation about required minimum effective block size if someone decides not to use OS_MEMPOOL_BYTES() macro to get proper size.

But even without such note for  block_size<4 user needs to assume at least 4 bytes for each block anyway - documentation clearly states that effective block size is OS_ALIGN-ed so anything less than 4 bytes is aligned to 4 bytes for all archs now (since we have only 32-bit archs).